jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/di/DependencyInjectionDiagnosticsCollector.java

```diff
@@ -84,9 +113,33 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
 	                }    
 	            }    
 	        }
+	        for (IType type : alltypes) {
+				List<IMethod> constructorMethods = Arrays.stream(type.getMethods())
+	                    .filter(this::isConstructorMethod).collect(Collectors.toList());
+				
+				//there are no constructors
+				if(constructorMethods.size() == 0)
+					return;
+				boolean hasInjectConstructor = false;
+				int injectedConstructors = 0;
+				for (IMethod m : constructorMethods) {
+					hasInjectConstructor = Arrays.stream(m.getAnnotations())
+                    .map(annotation -> annotation.getElementName())
+                    .anyMatch(annotation -> annotation.equals("Inject"));
+					if (hasInjectConstructor) {
+						injectedConstructors++;
+					}
+					if(injectedConstructors > 1) {
+						 diagnostics.add(createDiagnostic(m,unit,"Inject cannot be used with multiple constructors",DIAGNOSTIC_CODE_INJECT_CONSTRUCTOR));
```

Comment:
As discussed the diagnostic should appear on all constructors with `@Inject` (if there are more than 1)